### PR TITLE
Fix issue where TargetedSweep couldn't sweep tables with no namespace

### PR DESCRIPTION
**Pull request by dxiao**
Like tables from a certain large project. :P

**Goals (and why)**:

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3532)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/api/WriteReference.java [hunk 1]
```diff
@@ -56,13 +56,14 @@ public CellReference cellReference() {
...
 
+    @SuppressWarnings("deprecated")
     public static final Hydrator<WriteReference> BYTES_HYDRATOR = input -> {
         if (input[0] != prefix[0]) {
             return decodeLegacy(input);
         }
         int offset = 1;
         String tableReferenceString = EncodingUtils.decodeVarString(input, offset);
-        TableReference tableReference = TableReference.createFromFullyQualifiedName(tableReferenceString);
+        TableReference tableReference = TableReference.createUnsafe(tableReferenceString);
```
Comment (j-baker) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/api/WriteReference.java]: I'm not even sure this is right. How about 'fromString'? This class is full of juicy code, e.g.

createUnsafe vs fromString vs fromInternalTableName all having slightly different implementations

Comment (j-baker) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/api/WriteReference.java]: Do we need to worry about fromInternalTableName and the curse of __ -> ., basically?

Comment (j-baker) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/api/WriteReference.java]: turns out no, that's a Cassandra thing

Comment (j-baker) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/api/WriteReference.java]: but still - maybe fromString, since it's not deprecated?

**Comment by dxiao** [comment]
Also, can somebody explain to me what is going on with the test failure? 
```
[00:05:15] Warning, treated as error:
[00:05:15] /home/circleci/project/docs/source/release_notes/release-notes.rst::Anonymous hyperlink mismatch: 1 references but 0 targets.
[00:05:15] See "backrefs" attribute for IDs.
[00:05:15] Makefile:56: recipe for target 'html' failed
```

**Comment by j-baker** [comment]
Fixed in #3533

**Pull request closed without merging**